core/embed/rust/src/ui/model_mercury/flow/confirm_output.rs

```diff
@@ -69,68 +113,302 @@ impl FlowState for ConfirmOutput {
+    let account_path: Option<TString> =
+        kwargs.get(Qstr::MP_QSTR_account_path)?.try_into_option()?;
+
+    let br_name: TString = kwargs.get(Qstr::MP_QSTR_br_name)?.try_into()?;
+    let br_code: u16 = kwargs.get(Qstr::MP_QSTR_br_code)?.try_into()?;
+
+    let address: Obj = kwargs.get(Qstr::MP_QSTR_address)?;
+    let amount: Option<Obj> = kwargs.get(Qstr::MP_QSTR_amount)?.try_into_option()?;
+
+    let chunkify: bool = kwargs.get_or(Qstr::MP_QSTR_chunkify, false)?;
+    let text_mono: bool = kwargs.get_or(Qstr::MP_QSTR_text_mono, true)?;
+
+    let summary_items: Obj = kwargs.get(Qstr::MP_QSTR_summary_items)?;
+    let fee_items: Obj = kwargs.get(Qstr::MP_QSTR_fee_items)?;
+
+    let summary_title: Option<TString> =
+        kwargs.get(Qstr::MP_QSTR_summary_title)?.try_into_option()?;
+    let summary_br_name: Option<TString> = kwargs
+        .get(Qstr::MP_QSTR_summary_br_name)?
+        .try_into_option()?;
+    let summary_br_code: Option<u16> = kwargs
+        .get(Qstr::MP_QSTR_summary_br_code)?
+        .try_into_option()?;
+
+    let cancel_text: Option<TString> = kwargs.get(Qstr::MP_QSTR_cancel_text)?.try_into_option()?;
+
+    // Address
+    let content_address = ConfirmBlobParams::new(TR::words__address.into(), address, None)
+        .with_subtitle(title)
+        .with_menu_button()
+        .with_footer(TR::instructions__swipe_up.into(), None)
+        .with_chunkify(chunkify)
+        .with_text_mono(text_mono)
+        .into_layout()?
+        .one_button_request(ButtonRequest::from_num(br_code, br_name));
+
+    // AddressMenu
+    let mut address_menu = VerticalMenu::empty();
+    let mut address_menu_items = Vec::<usize, 2>::new();
+    if account.is_some() && account_path.is_some() {
+        address_menu = address_menu.item(
+            theme::ICON_CHEVRON_RIGHT,
+            TR::address_details__account_info.into(),
+        );
+        unwrap!(address_menu_items.push(MENU_ITEM_ACCOUNT_INFO));
+    }
+    address_menu = address_menu.danger(
+        theme::ICON_CANCEL,
+        cancel_text.unwrap_or(TR::send__cancel_sign.into()),
+    );
+    unwrap!(address_menu_items.push(MENU_ITEM_CANCEL));
+    let content_address_menu = Frame::left_aligned(TString::empty(), address_menu)
+        .with_cancel_button()
+        .with_swipe(SwipeDirection::Right, SwipeSettings::immediate())
+        .map(move |msg| match msg {
+            FrameMsg::Content(VerticalMenuChoiceMsg::Selected(i)) => {
+                let selected_item = address_menu_items[i];
+                Some(FlowMsg::Choice(selected_item))
+            }
+            FrameMsg::Button(_) => Some(FlowMsg::Cancelled),
+        });
+
+    // AccountInfo
+    let ad = AddressDetails::new(TR::send__send_from.into(), account, account_path)?;
+    let content_account = ad.map(|_| Some(FlowMsg::Cancelled));
+
+    // AddressMenuCancel
+    let content_address_menu_cancel = Frame::left_aligned(
```

Comment:
All three `content_x_y_cancel` are identical and the code is copy-pasted 3x. We should write a function which constructs it.

Reply:
https://github.com/trezor/trezor-firmware/pull/4095/commits/ec6a2a8c8d7e840f3b19e2ab09c9bb08806c1449#diff-5cb317ed79ea16b58238d2743af510c9f82a27c2764811d83f532f8d8358b6f6R188